### DISCUSSION TITLE
How to use bulk action to set boolean property on multiple selected records

### DISCUSSION THREAD

**Discussion by LemonPie22** [filamentphp/filament#9725] [Help] [answered]
In my `parts` table I have a boolean `is_available` column. I'm trying to allow users to set this to `true` or `false` for the selected records.

My `PartResource` code has the following bulk action:
```
Tables\Actions\BulkAction::make('changeAvailability')
    ->label('Change availability')
    ->color('secondary')
    ->icon('heroicon-o-flag')
    ->action(function (Collection $records, array $data): void {
        foreach ($records as $record) {
            $record->available()->associate($data['is_available']);
            $record->save();
        }
    })
    ->form([
        Select::make('is_available')
            ->label('Available')
            ->options([true => 'True', false => 'False'])
            ->required(),
    ]),
```

My `Part` model has the following method:
```
/**
 * Get the availability of this part.
 */
public function available()
{
    return $this->is_available;
}
```

The `Change availability` option shows up under bulk actions:
<img width="398" alt="Screenshot 2023-11-17 at 23 52 04" src="https://github.com/filamentphp/filament/assets/107488019/dbd4758f-c365-4044-8616-10ba06121282">

And when it's clicked a modal form comes up showing `true` and `false`  as expected:
<img width="353" alt="Screenshot 2023-11-17 at 23 53 49" src="https://github.com/filamentphp/filament/assets/107488019/94fba388-91c8-47cd-9926-0b98ff527228">

But when I try to submit the modal form I get `Call to a member function associate() on bool` with reference to this line of code:
```
$record->available()->associate($data['is_available']);
```

I can see what's going wrong but don't know how to fix it. I'm guessing/hoping there's a much simpler method.

**Comment by coolsam726** [comment, marked as answer]
As you said, I am sure you can see what is wrong. You can only use `associate()` on relationships, but your `available()` method is not a relationship, it is just a getter. Just modify  this part:
```php
->action(function (Collection $records, array $data): void {
        foreach ($records as $record) {
            $record->available()->associate($data['is_available']);
            $record->save();
        }
    })
```
to this:
```php
->action(function (Collection $records, array $data): void {
        foreach ($records as $record) {
            $record->is_available = $data['is_available'];
            $record->save();
        }
    })
```
and you should be good to go.
